torch/csrc/autograd/function.h

```diff
@@ -257,11 +283,62 @@ struct TORCH_API Node : std::enable_shared_from_this<Node> {
   // Miscellaneous Methods
   //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-  /// The sequence number of this `Node`.
+  /// NOTE [ Sequence Number]
+  ///
+  /// The sequence_nr has two main usages in autograd:
+  ///
+  /// 1) Determine the node's execution priority in the engine.
+  ///    Nodes with higher priority numbers are executed first. There is actually no
+  ///    particular reason why we want nodes corresponding to ops executed later to be
+  ///    the first to be executed in the backward pass. The point is to simply make
+  ///    it *deterministic* so its a bit easier to reason about.
```

Comment:
Also that is important for some users. See the full note about reproducibility that actually guarantees determinism when a given flag is set. So the engine has to be deterministic!